app/validators/sections/section_validator.py

```diff
@@ -292,3 +294,97 @@ def _get_question_variant_fields_sets(variants):
+                        item_anchor_answer_id := item.get("item_anchor_answer_id")
+                    ) and (for_list := item.get("for_list")):
+                        items_map[for_list] = item_anchor_answer_id
+                        self._validate_item_anchor_answer_id_belongs_to_list_collector(
+                            items_map
+                        )
+
+                self._validate_related_answers_belong_to_list_collector(items)
+                self._validate_related_answer_has_label(items)
+
+    def _validate_related_answers_belong_to_list_collector(self, items):
+        for item in items:
+            if related_answers := item.get("related_answers"):
+                list_collector_ids = []
+                blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
+                for block in blocks:
+                    list_collector_ids.extend(
+                        iter(
+                            self.questionnaire_schema.get_list_collector_answer_ids(
+                                block["id"]
+                            )
+                        )
+                    )
+
+                for answer in related_answers:
+                    if answer["identifier"] not in list_collector_ids:
+                        self.add_error(
+                            error_messages.RELATED_ANSWERS_NOT_IN_LIST_COLLECTOR,
+                            id=answer["identifier"],
+                        )
+
+    def _validate_item_anchor_answer_id_belongs_to_list_collector(self, items_map):
+        list_collector_ids = set()
+
+        blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
+        for list_name, anchor_answer_id in items_map.items():
+            for block in blocks:
+                if block["for_list"] == list_name:
+                    list_collector_ids = (
+                        self.questionnaire_schema.get_list_collector_answer_ids(
+                            block["id"]
+                        )
+                    )
+
+            if anchor_answer_id not in list_collector_ids:
+                self.add_error(
+                    error_messages.ITEM_ANCHOR_ANSWER_ID_NOT_IN_LIST_COLLECTOR,
+                    id=anchor_answer_id,
+                )
+
+    def _validate_related_answer_has_label(self, items):
+        for item in items:
+            if related_answers := item.get("related_answers"):
+                for related_answer in related_answers:
+                    answer = self.questionnaire_schema.get_answer(
+                        related_answer["identifier"]
+                    )
+                    if not answer.get("label"):
+                        self.add_error(
+                            error_messages.NO_LABEL_FOR_RELATED_ANSWER.format(
+                                answer_id=related_answer["identifier"],
+                            ),
+                            id=related_answer["identifier"],
+                        )
```

Comment:
The duplication and overall validation can be simplified with something like:
```suggestion
    def validate_section_summary(self):
        summary_items = self.schema_element.get("summary", {}).get("items", [])
        if not summary_items:
            return None

        blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
        list_collector_answer_ids_by_list = defaultdict(list)
        for block in blocks:
            list_collector_answer_ids_by_list["for_list"].append(
                self.questionnaire_schema.get_list_collector_answer_ids(block["id"])
            )

        for item in summary_items:
            list_collector_answer_ids_for_list = list_collector_answer_ids_by_list[
                item["for_list"]
            ]

            if item_anchor_answer_id := item.get("item_anchor_answer_id"):
                self._validate_item_anchor_answer_id_belongs_to_list_collector(
                    item_anchor_answer_id, list_collector_answer_ids_for_list
                )

            for answer_source in item.get("related_answers", []):
                self._validate_related_answer_belong_to_list_collector(
                    answer_source, list_collector_answer_ids_for_list
                )
                self._validate_related_answer_has_label(answer_source)

    def _validate_related_answer_belong_to_list_collector(
        self, answer_source, list_collector_answer_ids
    ):
        if answer_source["identifier"] not in list_collector_answer_ids:
            self.add_error(
                error_messages.RELATED_ANSWERS_NOT_IN_LIST_COLLECTOR,
                id=answer_source["identifier"],
            )

    def _validate_item_anchor_answer_id_belongs_to_list_collector(
        self, anchor_answer_id, list_collector_answer_ids
    ):
        if anchor_answer_id not in list_collector_answer_ids:
            self.add_error(
                error_messages.ITEM_ANCHOR_ANSWER_ID_NOT_IN_LIST_COLLECTOR,
                id=anchor_answer_id,
            )

    def _validate_related_answer_has_label(self, answer_source):
        answer = self.questionnaire_schema.get_answer(answer_source["identifier"])
        if not answer.get("label"):
            self.add_error(
                error_messages.NO_LABEL_FOR_RELATED_ANSWER.format(
                    answer_id=answer_source["identifier"],
                ),
                id=answer_source["identifier"],
            )
```

Reply:
I suggest renaming `validate_section_summary` to `validate_section_summary_items`.

Reply:
Added now 👍